### PR TITLE
xds: adding more delta protocol tests

### LINKED ISSUE #10125: add more incremental xDS protocol tests
Add additional xDS protocol tests for #9855

### PR DESCRIPTION
**Pull request by rboyer**
Fixes #10125

### CODE REVIEW

File: agent/xds/delta_test.go [hunk 1]
```diff
@@ -45,132 +37,228 @@ func TestServer_DeltaAggregatedResources_v3_BasicProtocol_TCP(t *testing.T) {
...
-	// And no other response yet
-	assertDeltaChanBlocked(t, envoy.deltaStream.sendCh)
+	deleteAllButOneEndpoint := func(snap *proxycfg.ConfigSnapshot, svc, targetID string) {
+		snap.ConnectProxy.ConfigSnapshotUpstreams.WatchedUpstreamEndpoints[svc][targetID] =
+			snap.ConnectProxy.ConfigSnapshotUpstreams.WatchedUpstreamEndpoints[svc][targetID][0:1]
+	}
 
-	// ACKs the listener
-	envoy.SendDeltaReqACK(t, ListenerType, 3, true, nil)
+	runStep(t, "envoy randomly decides to unsubscribe from an endpoint", func(t *testing.T) {
```
Comment (freddygv) [inline agent/xds/delta_test.go:171]: I think renaming the subtest to include the expected result could make the test case clearer:
```suggestion
	runStep(t, "avoid sending config for unsubscribed resource", func(t *testing.T) {
```

File: agent/xds/testing.go [hunk 2]
```diff
@@ -214,12 +253,21 @@ func (e *TestEnvoy) SendDeltaReqACK(
 	ack bool,
 	errorDetail *status.Status,
 ) {
+	// TODO(rb): fix this signature since we have an ACK/NACK pair of methods.
```
Comment (freddygv) [inline agent/xds/testing.go:256]: TODO for this PR?

**Review by freddygv — APPROVED** [review]
LGTM, added a couple non-blocking comments below

File: agent/xds/testing.go (hunk 2 above)
Comment (rboyer) [reply agent/xds/testing.go]: TODONE with the latest commit

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/387485.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 80c39f1083d68ab5f91c7fc78d66b0e21a09fb28 onto `release/1.10.x` succeeded!

**Pull request merged**